src/ddsrt/src/event.c

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright(c) 2006 to 2020 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include "dds/ddsrt/event.h"
+#include "dds/ddsrt/event_pipe.h"
+
+/**
+* ddsrt_event creation function
+*
+* mon_type: type of object being monitored
+* ptr_to_mon: pointer to object being monitored
+* mon_sz: size in bytes of object being monitored
+* evt_type: type of event being monitored for
+*/
+struct ddsrt_event* ddsrt_event_create(enum MONITORABLE_TYPE mon_type, void* ptr_to_mon, unsigned int mon_sz, int evt_type) {
+    if (0 == mon_sz ||
+        sizeof(void*) < mon_sz) return NULL;
+    if (NULL == ptr_to_mon &&
+        MONITORABLE_UNSET != mon_type) return NULL;
+    struct ddsrt_event* returnptr = malloc(sizeof(struct ddsrt_event));
+    if (NULL == returnptr) return NULL;
+
+    void* tmpptr = malloc(8);
```

Comment:
Please use the `ddsrt_` family of function where possible. In this case particularly: While `ddsrt_malloc` is mapped directly to `malloc` on most platforms, that's not true everywhere. e.g. on FreeRTOS the two are very different.